### PR TITLE
Complete Level 3 - 4 and Add Lost Field "TankCapacity" To Vehicle  

### PR DESCRIPTION
**Pull request by UNDogMan**

### CODE REVIEW

File: Pre-DevIncubator/Models/Engine/AbstractEngine.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pre_DevIncubator.Models.Engine
+{
+    public abstract class AbstractEngine
+    {
+        public string Model { get; init; }
+        public double TaxCoefficient { get; set; }
```
Comment (PushMeNow) [inline Pre-DevIncubator/Models/Engine/AbstractEngine.cs]: Why TaxCoefficient doesn't have init keyword but Model has?

File: Pre-DevIncubator/Models/Engine/DieselEngine.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pre_DevIncubator.Models.Engine
+{
+    public class DieselEngine : CombustionEngine
+    {
+        public DieselEngine(double engineCapacity, double fuelConsumptionPer100, int tankCapacity)
+            :base("Diesel", 1.2)
```
Comment (PushMeNow) [inline Pre-DevIncubator/Models/Engine/DieselEngine.cs]: Space between ':' and 'base'

File: Pre-DevIncubator/Models/Engine/ElectricalEngine.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Pre_DevIncubator.Models.Engine
+{
+    public class ElectricalEngine : AbstractEngine
+    {
+        public double ElectricityConsumption { get; init; }
+        public int HorsePowers { get; set; }
+
+        public ElectricalEngine(double electricityConsumption, int battaryCapacity):
+            base("Electrical", 0.1)
```
Comment (PushMeNow) [inline Pre-DevIncubator/Models/Engine/ElectricalEngine.cs]: Could you configure your IDE formatter correctly? Why is there not space before ':' ?

File: Pre-DevIncubator/Models/Engine/ElectricalEngine.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pre_DevIncubator.Models.Engine
+{
+    public class ElectricalEngine : AbstractEngine
+    {
+        public double ElectricityConsumption { get; init; }
```
Comment (PushMeNow) [inline Pre-DevIncubator/Models/Engine/ElectricalEngine.cs]: Again I don't realize why here init keyword

File: Pre-DevIncubator/Models/Vehicle.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        public VehicleType VehicleType { get; init; }
+        public AbstractEngine Engine { get; init; }
+        public string ModelName { get; init; }
+        public string RegistrationNumber { get; set; }
+        public int Weight { get; set; }
+        public int ManufactureYear { get; init; }
+        public int Mileage { get; set; }
+        public Color Color { get; set; }
+        public double TankCapacity { get; set; }
```
Comment (PushMeNow) [inline Pre-DevIncubator/Models/Vehicle.cs]: init?

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi!
I did some comments. Let me know if any comments

**Review by PushMeNow — APPROVED** [review]
Looks good! Approve.

**Review by dbogdan2000 — APPROVED** [review]

**Review by ArtyomKolosov2 — APPROVED** [review]

**Pull request merged**